### PR TITLE
PILOT-1990: Add support for project announcements route

### PR DESCRIPTION
**Pull request by vmoshynskyi**
## Summary

Add support for project announcements route

## JIRA Issues

PILOT-1990

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

### CODE REVIEW

File: api/api_announcement/announcement.py [hunk 1]
```diff
@@ -86,13 +89,37 @@ async def post(self, request: Request):
...
+@router.get(
+    '/project/{project_code}/announcements/',
+    summary='List project announcements',
+    dependencies=[Depends(PermissionsCheck('announcement', '*', 'view'))],
+)
+async def list_project_announcements(
+    request: Request,
+    project: ProjectObject = Depends(get_project),
+    notification_service_client: NotificationServiceClient = Depends(get_notification_service_client),
+):
```
Comment (afmam) [inline api/api_announcement/announcement.py:101]: Missing docstring and return type hint

File: api/api_announcement/announcement.py [hunk 2]
```diff
@@ -86,13 +89,37 @@ async def post(self, request: Request):
...
+    '/project/{project_code}/announcements/',
+    summary='Create project announcement',
+    dependencies=[Depends(PermissionsCheck('announcement', '*', 'create'))],
+)
+async def create_project_announcement(
+    body: ProjectAnnouncementCreateSchema,
+    project: ProjectObject = Depends(get_project),
+    current_identity: Dict[str, Any] = Depends(jwt_required),
+    notification_service_client: NotificationServiceClient = Depends(get_notification_service_client),
+):
```
Comment (afmam) [inline api/api_announcement/announcement.py:117]: Missing docstring and return type hint

**Review by afmam — CHANGES_REQUESTED** [review]
Left two comments

**Comment by vmoshynskyi** [comment]
I added a commit with fixes. Please check.

**Review by afmam — APPROVED** [review]
LGTM

**Pull request merged**